plugins/generic/pln/classes/DepositPackage.inc.php

```diff
@@ -84,6 +84,32 @@ function getAtomDocumentPath() {
 	 function getPackageFilePath() {
 		return $this->getDepositDir() . DIRECTORY_SEPARATOR . $this->_deposit->getUUID() . ".zip";
 	}
+	
+	/**
+	 * Create a DOMElement in the $dom, and set the element name, namespace, and 
+	 * content appropriately. Named HTML entities in the content are converted
+	 * to UTF-8 characters. Any invalid UTF-8 characters will be dropped.
+	 * 
+	 * @param DOMDocument $dom
+	 * @param string $elementName
+	 * @param string $content
+	 * @param string $namespace
+	 * @return DOMElement
+	 */
+	function _generateElement($dom, $elementName, $content, $namespace = null){
+		// remove any invalid UTF-8.
+		$original = mb_substitute_character();
+		mb_substitute_character(0xFFFD);
+		$filtered = mb_convert_encoding($content, 'UTF-8', 'UTF-8');
+		mb_substitute_character($original);
+		
+		// put the filtered content in a CDATA, as it may contain markup that 
+		// isn't valid XML.
+		$node = $dom->createCDATASection($filtered);
+		$element = $dom->createElementNS($namespace, $elementName);
```

Comment:
Does `createElementNS()` handle a null namespace?  I don't see that case in the PHP docs.